src/Helmet.jsx

```diff
@@ -265,6 +272,8 @@ const handleClientStateChange = (newState) => {
     updateTags(TAG_NAMES.META, metaTags);
     updateTags(TAG_NAMES.BASE, baseTag);
 
+    onChangeClientState(newState);
```

Comment:
Should this be more selective about which properties of `newState` are getting passed to the callback? Something like:

```javascript
onChangeClientState({title, baseTag, metaTags, linkTags, scriptTags});
```